### PR TITLE
Add Version Sort

### PR DESCRIPTION
**Pull request by juansc**
# Description

Addresses https://github.com/lsd-rs/lsd/issues/801. The root cause is that the external library that is being used does not have sufficient test coverage and is no longer maintained.

I took a look at how [GNU sort is implemented](https://github.com/coreutils/coreutils/blob/master/doc/sort-version.texi) and it looked simple enough to be included into this repo. I added a few test cases, including the one from the issue. 

I'm still pretty new to Rust, so any code/organizational feedback is greatly appreciated

---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests (You tell me 🙂)
- [x] Add changelog entry
- [x] Update default config/theme in README (if applicable)
- [x] Update man page at lsd/doc/lsd.md (if applicable)

### CODE REVIEW

**Comment by juansc** [comment]
It looks like I have some refactoring artifacts. I'll fix and add more documentation

File: src/sort.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use crate::flags::{DirGrouping, Flags, SortColumn, SortOrder};
 use crate::meta::Meta;
-use human_sort::compare;
+use crate::versionsort::compare;
```
Comment (nim65s) [inline src/sort.rs:3]: I think you didn't meant to simply remove human-sort, but this is what you are doing here. 

Did you forget to implement some cli / config switch ?

**Review by nim65s — CHANGES_REQUESTED** [review]

**Comment by juansc** [comment]
I think it's cleaner to break out the sorting logic into a [separate crate](https://github.com/juansc/vsort/). Once it's published I'll update this PR to use that crate instead. The code in the create has been updated and has more robust testing

File: src/sort.rs (hunk 1 above)
Comment (juansc) [reply src/sort.rs]: I do mean to replace the sorting library. The new code is now its own separate crate and the PR now reflects that.

**Comment by zwpaper** [comment]
hi @juansc, I have read the PR and vsort crate, and it seems great!

please add a changelog here https://github.com/lsd-rs/lsd/blob/master/CHANGELOG.md and we should be good to go.

/lgtm

**Comment by juansc** [comment]
@zwpaper Changelog has been added

**Comment by zwpaper** [comment]
thanks so much @juansc

**Pull request merged**